### PR TITLE
Added a check if minimum required gluster version available

### PR DESCRIPTION
**Pull request by shtripat**
tendrl-bug-id: Tendrl/gluster-integration#161
Signed-off-by: Shubhendu <shtripat@redhat.com>

### CODE REVIEW

**Comment by shtripat** [comment]
@Tendrl/tendrl-core plz review

File: tendrl/commons/flows/import_cluster/__init__.py [hunk 1]
```diff
@@ -86,6 +86,20 @@ def run(self):
...
         else:
+            # Check if minimum required version of underlying gluster
+            # cluster met. If not fail the import task
+            sds_version = NS.etcd_orm.client.read(
+                "nodes/%s/DetectedCluster/sds_pkg_version" %
+                NS.node_context.node_id
+            ).value
+            maj_ver, min_ver, rel = sds_version.split('.')
+            req_maj_ver, req_min_ver, req_rel = \
+                NS.config.data['min_reqd_gluster_ver'].split('.')
```
Comment (r0h4n) [inline tendrl/commons/flows/import_cluster/__init__.py]: How is a user config item going to help? The user can change this to anything he wants.

This should be in the definitions here https://github.com/Tendrl/commons/blob/develop/tendrl/commons/objects/definition/master.yaml

Comment (shtripat) [inline tendrl/commons/flows/import_cluster/__init__.py]: Ok. Will add one data point here. Can you guide me to the section where this should be added.

File: tendrl/commons/flows/import_cluster/__init__.py [hunk 2]
```diff
@@ -86,6 +86,29 @@ def run(self):
             if is_mon:
                 import_ceph(NS.tendrl_context.integration_id)
         else:
+            # Check if minimum required version of underlying gluster
+            # cluster met. If not fail the import task
+            sds_version = NS.etcd_orm.client.read(
```
Comment (r0h4n) [inline tendrl/commons/flows/import_cluster/__init__.py]: Please Create the DetectedCluster object and load it, stop using etcd_orm directly for no reason, you do realize that we loose all the base object functions when you do a direct etcd_orm.client.read()

eg: detected_cluster  = NS.tendrl.objects.DetectedCluster().load()
detected_cluster.sds_version....

File: tendrl/commons/objects/definition/master.yaml [hunk 3]
```diff
@@ -439,4 +439,6 @@ namespace.tendrl:
       help: "Platform of the Node"
       value: nodes/$Node_context.node_id/Platform
       list: nodes/$Node_context.node_id/Platform
-tendrl_schema_version: 0.3
\ No newline at end of file
+tendrl_schema_version: 0.3
+min_reqd_gluster_ver: 3.9.0
```
Comment (r0h4n) [inline tendrl/commons/objects/definition/master.yaml]: Put this inside a namespace man, how are we going to map this inside tendrl?

Please define these inside the "namespace.tendrl"

File: tendrl/commons/flows/import_cluster/__init__.py (hunk 2 above)
Comment (shtripat) [inline tendrl/commons/flows/import_cluster/__init__.py]: ok

File: tendrl/commons/objects/definition/master.yaml (hunk 3 above)
Comment (shtripat) [inline tendrl/commons/objects/definition/master.yaml]: ok

**Comment by r0h4n** [comment]
@shtripat Verified?

**Pull request merged**